File: javascript/1.MergeAlternately.js
function mergeAlternately(word1, word2) {

    let charactersWord1 = word1.split("")
    let charactersWord2 = word2.split("")

    let length = Math.max(charactersWord1.length, charactersWord2.length)
    let merged = ""

    for(let i = 0; i < length; i++){
        if(charactersWord1.length > i) merged += charactersWord1[i]
        if(charactersWord2.length > i) merged += charactersWord2[i]

    }

    return merged;

}

console.log(mergeAlternately('abc', 'ZPE'));
console.log(mergeAlternately('world', 'HELLO'));
console.log(mergeAlternately('number', 'CORRIENDO'));
console.log(mergeAlternately('coding', 'EXACTLY'));
console.log(mergeAlternately('working', 'TREE'));

File: python/1.MergeAlternately.py
def mergeAlternately(word1, word2):
    
    charactersWord1 = list(word1)
    charactersWord2 = list(word2)

    length = max([len(charactersWord1), len(charactersWord2)])
    merged = ""

    for i in range(length):
        if len(charactersWord1) > i:
            merged += charactersWord1[i]
        if len(charactersWord2) > i:
            merged += charactersWord2[i]



    return merged


print(mergeAlternately("abc", "ZPE"))
print(mergeAlternately("world", "HELLO"))
print(mergeAlternately("number", "CORRIENDO"))
print(mergeAlternately("coding", "EXACTLY"))
print(mergeAlternately("working", "TREE"))
